src/MongoDB.Driver/MongoDatabaseImpl.cs

```diff
@@ -396,12 +424,22 @@ private DropCollectionOperation CreateDropCollectionOperation(string name)
             };
         }
 
+        private ListCollectionsOperation CreateListCollectionNamesOperation(ListCollectionNamesOptions options)
+        {
+            var messageEncoderSettings = GetMessageEncoderSettings();
+            return new ListCollectionsOperation(_databaseNamespace, messageEncoderSettings)
+            {
+                Filter = options?.Filter?.Render(_settings.SerializerRegistry.GetSerializer<BsonDocument>(), _settings.SerializerRegistry),
+                NameOnly = true
+            };
+        }
+
         private ListCollectionsOperation CreateListCollectionsOperation(ListCollectionsOptions options)
         {
             var messageEncoderSettings = GetMessageEncoderSettings();
             return new ListCollectionsOperation(_databaseNamespace, messageEncoderSettings)
             {
-                Filter = options.Filter?.Render(_settings.SerializerRegistry.GetSerializer<BsonDocument>(), _settings.SerializerRegistry)
+                Filter = options?.Filter?.Render(_settings.SerializerRegistry.GetSerializer<BsonDocument>(), _settings.SerializerRegistry)
```

Comment:
Is this fixing an existing bug?  If so, does it warrant a unit test?

Reply:
Not an existing bug. The callers used to check for null and create a default options if necessary. That's an unnecessary extra memory allocation. Now callers are allowed to pass null through to this method.

But calling ListCollectionNames and ListCollections with null options should definitely be tested. I've modified the existing tests to also test with null options.